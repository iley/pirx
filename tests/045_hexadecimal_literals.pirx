func main(): int {
  var x: int;
  var y: int64;
  
  // Test 32-bit hex literals
  x = 0x42;
  printf("x = %d\n", x);
  
  x = 0X10;
  printf("x = %d\n", x);
  
  x = 0xABCD;
  printf("x = %d\n", x);
  
  x = 0xabcd;
  printf("x = %d\n", x);
  
  x = 0x0;
  printf("x = %d\n", x);
  
  // Test 64-bit hex literals
  y = 0x42l;
  printf("y = %lld\n", y);
  
  y = 0xdeadbeefl;
  printf("y = %lld\n", y);
  
  y = 0X1234567890ABCDEFl;
  printf("y = %lld\n", y);
  
  return 0;
}