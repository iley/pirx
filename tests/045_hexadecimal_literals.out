x = 66
x = 16
x = 43981
x = 43981
x = 0
y = 66
y = 3735928559
y = 1311768467294899695
